sql/plan/showvariables.go

```diff
@@ -90,7 +92,20 @@ func (sv *ShowVariables) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, er
 		)
 	}
 
-	for k, v := range ctx.GetAllSessionVariables() {
+	filteredMap := make(map[string]interface{})
+	var err error
+
+	sessVariables := ctx.GetAllSessionVariables()
+	if sv.filter != nil {
+		filteredMap, err = filterVariables(ctx, sessVariables, sv.filter)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		filteredMap = sessVariables
+	}
+
+	for k, v := range filteredMap {
```

Comment:
if you combine the like and filter into one expression this eval would be a bit simpler